src/main/java/org/apache/commons/math4/util/MathArrays.java

```diff
@@ -73,16 +73,16 @@ private MathArrays() {}
          * @param array Array to operate on.
          * @return the result of the operation.
          */
-        double evaluate(double[] array);
+        double evaluate(double @MinLen(1) [] array);
```

Comment:
Why is array `@MinLen(1)`? (Here and in similar methods throughout this PR)? Some methods I looked at return 0 or NaN if the array is empty.